astropy/io/ascii/ipac.py

```diff
@@ -214,6 +216,10 @@ def get_cols(self, lines):
                 null = header_vals[3][i].strip()
                 fillval = "" if issubclass(col.type, core.StrType) else "0"
                 self.data.fill_values.append((null, fillval, col.name))
+            if "long".startswith(col.raw_type.lower()):
+                # ensure long columns are 64-bit int, to address (Windows-specific):
```

Comment:
The comment specifically targets Windows but I don't see a `sys.platform` check in the `if`.

Reply:
The issue surfaces on Windows. Conceivably, the same problem could arise in some  other 32-bit platform, or possibly some non-standard build of `numpy` [*].

I think mapping `long` to `np.int64` should be safe and correct in all cases.

Thoughts?

---

[*] When typing is not specified, `ascii` guesses and assigns type for a column in [`_convert_vals()`](https://github.com/astropy/astropy/blob/d1579ecbce54be7aa22dcf7cfb17b6cb1c537fbe/astropy/io/ascii/core.py#L1097) .

For integer-like columns, it uses the converter from [`convert_numpy(int)`](https://github.com/astropy/astropy/blob/d1579ecbce54be7aa22dcf7cfb17b6cb1c537fbe/astropy/io/ascii/core.py#L977), which boils down to the behavior of `np.array([], int)`. For Windows (even on 64-bit), `numpy` returns a `np.int32` array, while for (most if not all) other platforms, it returns a `np.int64` array.

Reply:
If your actual patch is not really platform specific, then perhaps update the comment and change log to match? Currently, from the change log, it seems like your fix is only for Windows though in reality it is not confined to Windows. Hope that makes sense. Thanks!

Reply:
For changelog (and similarly in the comment), how about: 
```rst
Fixed reading IPAC tables for ``long`` column type on some platforms, e.g., Windows.
```

Reply:
That is fine by me. Thanks for your understanding!

Reply:
Done.